cmd/restic/cmd_backup.go

```diff
@@ -550,6 +555,15 @@ func runBackup(opts BackupOptions, gopts GlobalOptions, term *termstatus.Termina
 	}
 
 	var targetFS fs.FS = fs.Local{}
+	if runtime.GOOS == "windows" && opts.UseWindowsVSS {
+		if !fs.HasSufficientPrivilegesForVSS() {
+			return errors.Fatal("user doesn't have sufficient privileges to use VSS snapshots\n")
+		}
+
+		localVss := fs.NewLocalVss(p.E, p.P, p.V)
```

Comment:
Is it OK to call ArchiveProgressReporter.Error() without any information besides the error? What are the expectations towards the caller?

Reply:
internal/ui/backup.go only uses `err`, internal/ui/json/backup.go uses `item` and `err`. For the call in the archiver it looks like `item` and `err` are always filled with reasonable values, whereas `fi` could also receive an uninitialized os.FileInfo object (and is not used anyways). So if there's a filename at hand then please pass it into `item` and if there is none just pass an empty string `""`.

Reply:
I've started to implement using Error() instead of E() for error reporting. This is not committed yet because I got stuck handling normal messages. What about messages like "successfully created snapshot for ..." etc.?

Reply:
The current solution is that such informative messages are not printed when JSON output is enabled. So you can pass just pass a method which drop these message when JSON output is enabled. That way it's rather easy to maybe change this in the future.

Reply:
Fixed in https://github.com/restic/restic/pull/2274/commits/0831d98d80b2db6c8fe9b88407da987f10a15d59